fulp_modules/ipc/code/surgery/body_repair.dm

```diff
@@ -0,0 +1,128 @@
+			if(!..())
+				break
+
+/datum/surgery_step/robot_heal/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/umsg = "You succeed in fixing some of [target]'s damage" //no period, add initial space to "addons"
+	var/tmsg = "[user] fixes some of [target]'s damage" //see above
+	var/urhealedamt_brute = 0
+	if(healsbrute)
+		urhealedamt_brute = brutehealing
+		tool.use_tool(target, user, 0, volume=50, amount=1)
+	var/urhealedamt_burn = 0
+	if(healsburn)
+		urhealedamt_burn = burnhealing
+		if(tool)
+			tool.use(1)
+	if(missinghpbonus)
+		if(target.stat != DEAD)
+			urhealedamt_brute += round((target.getBruteLoss()/ missinghpbonus),0.1)
+			urhealedamt_burn += round((target.getFireLoss()/ missinghpbonus),0.1)
+		else //less healing bonus for the dead since they're expected to have lots of damage to begin with (to make TW into defib not TOO simple)
+			urhealedamt_brute += round((target.getBruteLoss()/ (missinghpbonus*5)),0.1)
+			urhealedamt_burn += round((target.getFireLoss()/ (missinghpbonus*5)),0.1)
+	if(!get_location_accessible(target, target_zone))
+		urhealedamt_brute *= 0.55
+		urhealedamt_burn *= 0.55
+		umsg += " as best as you can while they have clothing on"
+		tmsg += " as best as they can while [target] has clothing on"
+	target.heal_bodypart_damage(urhealedamt_brute,urhealedamt_burn, 0, BODYPART_ROBOTIC)
+	display_results(user, target, "<span class='notice'>[umsg].</span>",
+		"[tmsg].",
+		"[tmsg].")
+	if(istype(surgery, /datum/surgery/robot_healing))
+		var/datum/surgery/robot_healing/the_surgery = surgery
+		the_surgery.antispam = TRUE
+	return TRUE
+
+/datum/surgery_step/robot_heal/failure(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	display_results(user, target, "<span class='warning'>You screwed up!</span>",
+		"<span class='warning'>[user] screws up!</span>",
+		"<span class='notice'>[user] fixes some of [target]'s damage.</span>", TRUE)
+	var/urdamageamt_brute = 0
+	if(healsbrute)
+		urdamageamt_brute = brutehealing * 0.8
+	var/urdamageamt_burn = 0
+	if(healsburn)
+		urdamageamt_burn = burnhealing * 0.8
+	if(missinghpbonus)
+		urdamageamt_brute += round((target.getBruteLoss()/ (missinghpbonus*2)),0.1)
+		urdamageamt_burn += round((target.getFireLoss()/ (missinghpbonus*2)),0.1)
+
+	target.take_bodypart_damage(urdamageamt_brute, urdamageamt_burn)
+	return FALSE
+
+/***************************TYPES***************************/
+/datum/surgery/robot_healing/basic
+	name = "Repair robotic limbs (basic)"
+	healing_step_type = /datum/surgery_step/robot_heal/basic
+	desc = "A surgical procedure that provides repairs and maintenance to robotic limbs. Is slightly more efficient when the patient is severely damaged."
+	replaced_by = null
+
+/***************************STEPS***************************/
+
+/datum/surgery_step/robot_heal/basic
+	name = "repair damage"
+	brutehealing = 10
+	burnhealing = 10
+	missinghpbonus = 15
```

Comment:
will this show up alongside the abstract version of this surgery?

also, why would I ever do this surgery when just directly applying a welder or cable coils would heal an IPC faster and more efficiently (IIRC) and there are no ghetto alternatives to using a welder or cable coils?